float64[] heat_values
